File: Kotlin/프로그래머스/1/135808. 과일 장수/과일 장수.kt
class Solution {
    fun solution(k: Int, m: Int, score: IntArray): Int {
        var answer: Int = 0

        val sortedDesc = score.sortedDescending().chunked(m)

        for (intList in sortedDesc) {
            if (intList.size == m) {
                val min = intList[m-1]
                answer += min * m
            }
        }

        return answer
    }
}

File: C#/프로그래머스/1/135808. 과일 장수/과일 장수.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int k, int m, int[] score) {
        int answer = 0;
        List<int> scoreList = new List<int>(score);
        scoreList.Sort((a,b) => b.CompareTo(a));
        
        for(int i = 0; i < m - (scoreList.Count % m); i++)
        {
            scoreList.Add(0);
        }
        
        for(int i = m - 1; i < scoreList.Count; i += m)
        {
            answer += scoreList[i] * m;
        }
        
        
        return answer;
    }
}